src/plugins/memdump/memdump.cpp

```diff
@@ -0,0 +1,314 @@
+        }
+    }
+
+    fclose(fp);
+    ret = true;
+
+printout:
+    switch (plugin->m_output_format)
+    {
+        case OUTPUT_CSV:
+            printf("memdump," FORMAT_TIMEVAL ",%" PRIu32 ",0x%" PRIx64 ",\"%s\",%" PRIi64 ",\"%s\",\"%s\",%d,%" PRIx64 ",%" PRIu64 ",\"%s\"\n",
+                   UNPACK_TIMEVAL(info->timestamp), info->vcpu, info->regs->cr3, info->proc_data.name,
+                   info->proc_data.userid, info->trap->name, reason, info->proc_data.pid, ctx->addr, num_pages * VMI_PS_4KB, display_file);
+            break;
+        case OUTPUT_KV:
+            printf("memdump Time=" FORMAT_TIMEVAL ",PID=%d,PPID=%d,ProcessName=\"%s\",Method=%s,DumpReason=\"%s\",DumpPID=%d,DumpAddr=%" PRIx64 ",DumpSize=%" PRIu64 ",DumpFilename=\"%s\"\n",
+                   UNPACK_TIMEVAL(info->timestamp), info->proc_data.pid, info->proc_data.ppid, info->proc_data.name,
+                   info->trap->name, reason, info->proc_data.pid, ctx->addr, num_pages * VMI_PS_4KB, display_file);
+            break;
+        case OUTPUT_JSON:
+            escaped_pname = drakvuf_escape_str(info->proc_data.name);
+            escaped_fname = drakvuf_escape_str(display_file);
+            printf( "{"
+                    "\"Plugin\": \"memdump\","
+                    "\"TimeStamp\":" "\"" FORMAT_TIMEVAL "\","
+                    "\"ProcessName\": %s,"
+                    "\"UserName\": \"%s\","
+                    "\"UserId\": %" PRIu64 ","
+                    "\"PID\": %d,"
+                    "\"PPID\": %d,"
+                    "\"Method\": \"%s\","
+                    "\"DumpReason\": \"%s\","
+                    "\"DumpPID\": %d,"
+                    "\"DumpAddr\": %" PRIx64 ","
+                    "\"DumpSize\": %" PRIu64 ","
+                    "\"DumpFilename\": %s"
+                    "}\n",
+                    UNPACK_TIMEVAL(info->timestamp),
+                    escaped_pname,
+                    USERIDSTR(drakvuf), info->proc_data.userid,
+                    info->proc_data.pid, info->proc_data.ppid,
+                    info->trap->name, reason, info->proc_data.pid, ctx->addr,
+                    num_pages * VMI_PS_4KB, escaped_fname);
+            g_free(escaped_fname);
+            g_free(escaped_pname);
+            break;
+        default:
+        case OUTPUT_DEFAULT:
+            printf("[MEMDUMP] TIME:" FORMAT_TIMEVAL " VCPU:%" PRIu32 " CR3:0x%" PRIx64 ",\"%s\" %s:%" PRIi64" \"%s\" Reason:\"%s\" Process:%d Base:0x%" PRIx64 " Size:%" PRIu64 " File:\"%s\"\n",
+                   UNPACK_TIMEVAL(info->timestamp), info->vcpu, info->regs->cr3, info->proc_data.name,
+                   USERIDSTR(drakvuf), info->proc_data.userid, info->trap->name, reason, info->proc_data.pid, ctx->addr,
+                   num_pages * VMI_PS_4KB, display_file);
+            break;
+    }
+
+    printf("\n");
+
+done:
+    if (file)
+        free(file);
+
+    if (access_ptrs)
+        g_free(access_ptrs);
```

Comment:
Both `g_free` and `free` are null-safe, no need for the ifs.

Reply:
fixed